planner/core/task.go

```diff
@@ -1108,6 +1125,28 @@ func (p *PhysicalTopN) pushTopNDownToDynamicPartition(copTsk *copTask) (task, bo
 		tblScan = finalTblScanPlan.(*PhysicalTableScan)
 		tblInfo = tblScan.Table
 	}
+	if len(copTsk.idxMergePartPlans) > 0 {
```

Comment:
```suggestion
	if len(copTsk.idxMergePartPlans) > 0 {
	// calculate selectivities for each partial plan in advance and clone partial plans since we may modify their stats later.
```